PlayerBasePvpZones.cs

```diff
@@ -804,25 +886,236 @@ private IEnumerator CreateData()
+      Puts($"CreateDataForEnabledPlayers(): ...Startup completed in {(DateTime.UtcNow - startTime).TotalSeconds} seconds.");
 
       _createDataCoroutine = null;
     }
 
+    // Method to create zones for a specific player
+    private int CreatePlayerZones(ulong playerID)
+    {
+      Puts($"CreatePlayerZones(): Creating zones for player {playerID}...");
+      var count = 0;
+
+      // Create building zones
+      foreach (var (_, building) in BuildingManager.server.buildingDictionary)
+      {
+        var toolCupboard = GetToolCupboard(building);
+        if (!IsValid(toolCupboard) || !IsPlayerOwned(toolCupboard)) continue;
+        if (toolCupboard.OwnerID != playerID) continue;
+
+        CreateBuildingData(toolCupboard);
+        count++;
+      }
+
+      // Create shelter zones
+      if (LegacyShelter.SheltersPerPlayer.TryGetValue(playerID, out var shelterList))
+      {
+        foreach (var shelter in shelterList)
+        {
+          if (!IsValid(shelter) ||
+              !shelter.entityPrivilege.TryGet(true, out var legacyShelter) ||
+              !IsPlayerOwned(legacyShelter))
+          {
+            continue;
+          }
+          CreateShelterData(legacyShelter);
+          count++;
+        }
+      }
+
+      // Create tugboat zones - tugboats don't have direct owner tracking
+      // so we need to check each one
+      foreach (var serverEntity in BaseNetworkable.serverEntities)
+      {
+        if (serverEntity is not VehiclePrivilege tugboat || !IsValid(tugboat))
+        {
+          continue;
+        }
+
+        // Check if this player is authorized on the tugboat
+        if (!tugboat.authorizedPlayers.Exists(x => x.userid == playerID))
+        {
+          continue;
+        }
+
+        CreateTugboatData(tugboat);
+        count++;
+      }
+
+      Puts($"CreatePlayerZones(): Created {count} zone(s) for player {playerID}");
+      return count;
+    }
+
+    // Method to remove zones for a specific player
+    private void RemovePlayerZones(ulong playerID)
+    {
+      Puts($"RemovePlayerZones(): Removing zones for player {playerID}...");
+      var bulkDeleteList = Pool.Get<List<string>>();
+      var count = 0;
+
+      // Remove building zones
+      var buildingIDsToRemove = Pool.Get<List<NetworkableId>>();
+      foreach (var (tcID, buildingData) in _buildingData)
+      {
+        if (buildingData.ToolCupboard && buildingData.ToolCupboard.OwnerID == playerID)
+        {
+          buildingIDsToRemove.Add(tcID);
+        }
+      }
+      foreach (var tcID in buildingIDsToRemove)
+      {
+        DeleteBuildingData(tcID, bulkDeleteList);
+        count++;
+      }
+      Pool.FreeUnmanaged(ref buildingIDsToRemove);
+
+      // Cancel any pending building timers
+      CancelPlayerTimers(playerID, ref _buildingCheckTimers);
+      CancelPlayerTimers(playerID, ref _buildingCreateTimers);
+      CancelPlayerTimers(playerID, ref _buildingDeleteTimers);
```

Comment:
Feels like timers should always be taken care of first (not that it matters too much since RustDedicated is mostly single-threaded)